### PR TITLE
Add the `set translation_domain` inside the else clause

### PR DESCRIPTION
**Pull request by ejkun**
I am targeting this branch, because this PR will not cause a BC.

## Changelog
```Markdown
### Fixed
- Fixed the setting of the `translation_domain` twig variable. The value must change depending if the item has on_top set to true or false.
```
## Subject

The line with the `set translation_domain` must be called in the IF clause and in the ELSE clause. The IF will happen when the item being translated is inside a treeview, which means that the label must be translated against the translation_domain option. The ELSE will happen when on_top is true, which means that the label shoule be translated against the label_catalogue option.

### CODE REVIEW

File: Resources/views/Menu/sonata_menu.html.twig [hunk 1]
```diff
@@ -25,10 +25,11 @@
 
 {% block linkElement %}
     {% spaceless %}
-        {% set translation_domain = item.extra('label_catalogue', 'messages') %}
         {% if item.extra('on_top') is defined and not item.extra('on_top') %}
+            {% set translation_domain = item.extra('translation_domain') %}
             {% set icon = item.extra('icon')|default(item.level > 1 ? '<i class="fa fa-angle-double-right" aria-hidden="true"></i>' : '') %}
         {% else %}
+            {% set translation_domain = item.extra('label_catalogue') %}
```
Comment (OskarStark) [inline Resources/views/Menu/sonata_menu.html.twig]: the second argument `messages` isn't needed anymore?

**Comment by EmmanuelVella** [comment]
👍  currently side menu translations are broken !

**Comment by ejkun** [comment]
Sorry. It is, I will fix it ASAP.

Att.

Everton Barbosa Jr.
Bacharelando em Ciência da Computação
Fundação Universidade Federal do Tocantins

Em 17/08/2017 02:33, "Oskar Stark" <notifications@github.com> escreveu:

File: Resources/views/Menu/sonata_menu.html.twig (hunk 1 above)
Comment (EmmanuelVella) [inline Resources/views/Menu/sonata_menu.html.twig]: It is already defaulted here : https://github.com/sonata-project/SonataAdminBundle/pull/4601/files#diff-7aad62c164c47d66f90ce813f6f91c49L53

Anyway it's the default translation domain in symfony `Translator::trans()` method : https://github.com/symfony/translation/blob/master/Translator.php#L192

**Comment by peter-gribanov** [comment]
It's broken not only in the menu, but maybe it's problem in a different PR.

image

**Comment by OskarStark** [comment]
@ejkun please finish this PR, so we can create a release afterwards

**Comment by ejkun** [comment]
I will, as soon as I get to work.

**Comment by EmmanuelVella** [comment]
@OskarStark I meant that side menu translations are broken on the 3.x branch, this PR should fix it !

**Comment by ejkun** [comment]
I added the second argument back again. After this PR, the translation should be ok for items with on_top true or false.

File: Resources/views/Menu/sonata_menu.html.twig (hunk 1 above)
Comment (ejkun) [inline Resources/views/Menu/sonata_menu.html.twig]: I added it.

**Comment by ejkun** [comment]
@OskarStark is everything ok? Can you merge the PR?

**Comment by OskarStark** [comment]
Thank you @ejkun 👍

**Comment by gremo** [comment]
Can you please make a new release? Sidemenu is broken ATM. Thanks!

**Comment by OskarStark** [comment]
I already asked internally 👍

**Pull request merged**